dlt/common/libs/pyarrow.py

```diff
@@ -103,6 +101,83 @@ def table_name(self, value: str) -> None:
         self._update_message()
 
 
+class PyToArrowConversionException(Exception):
```

Comment:
please use `DltException` as base. People calling dlt methods should be able to filter exceptions coming from it.